documentation/modules/ROOT/pages/configuration/storage.adoc

```diff
@@ -0,0 +1,705 @@
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[jdbc-history-type]]<<jdbc-history-type, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.jdbc.history.JdbcSchemaHistory`.
+
+|[[jdbc-history-schema-history-url]]<<jdbc-history-schema-history-url, `schema.history.internal.jdbc.connection.url`>>
+|No default
+|JDBC driver connection string to connect to the database.
+
+|[[jdbc-history-schema-history-user]]<<jdbc-history-schema-history-user, `schema.history.internal.jdbc.connection.user`>>
+|No default
+|(Optional) Username of the database which will be used to access the database storage.
+
+|[[jdbc-history-schema-history-password]]<<jdbc-history-schema-history-password, `schema.history.internal.jdbc.connection.password`>>
+|No default
+|(Optional) Password of the database which will be used to access the database storage.
+
+|[[jdbc-history-wait-retry-delay-ms]]<<jdbc-history-wait-retry-delay-ms, `schema.history.internal.jdbc.connection.retry.delay.ms`>>
+|3 seconds
+|(Optional) Delay of retry on wait for connection failure.
+
+|[[jdbc-history-retry-max-attempts]]<<jdbc-history-retry-max-attempts, `schema.history.internal.jdbc.connection.retry.max.attempts`>>
+|5
+|(Optional) Maximum number of retry attempts before giving up.
+
+|[[jdbc-history-schema-history-table-name]]<<jdbc-history-schema-history-table-name, `schema.history.internal.jdbc.table.name`>>
+|`debezium_database_history`
+|The name of the table to which the history will be stored.
+
+|[[jdbc-history-schema-history-table-ddl]]<<jdbc-history-schema-history-table-ddl, `schema.history.internal.jdbc.table.ddl`>>
+|<<jdbc-history-schema-history-table-ddl-default, `see table section`>>
+|The DDL statement used to create the storage table.
+
+|[[jdbc-history-schema-history-table-select]]<<jdbc-history-schema-history-table-select, `schema.history.internal.jdbc.table.select`>>
+|<<jdbc-history-schema-history-table-select-default, `see table section`>>
+|The `SELECT` statement to read the internal schema history form the table.
+
+|[[jdbc-history-schema-history-table-exists-select]]<<jdbc-history-schema-history-table-exists-select, `schema.history.internal.jdbc.table.exists.select`>>
+|<<jdbc-history-schema-history-table-exists-select-default, `see table section`>>
+|The `SELECT` statement to check the existence of the storage table.
+
+|[[jdbc-history-schema-history-table-insert]]<<jdbc-history-schema-history-table-insert, `schema.history.internal.jdbc.table.insert`>>
+|<<jdbc-history-schema-history-table-insert-default, `see table section`>>
+|The `INSERT` statement to record the schema history to the table.
+
+|===
+
+==== Deprecated configuration prior to 3.2
+
+[cols="35%a,10%a,55%a",options="header"]
+|===
+|Property
+|Default
+|Description
+
+|[[deprecated-jdbc-history-type]]<<deprecated-jdbc-history-type, `schema.history.internal`>>
+|No default
+|Must be set to `io.debezium.storage.jdbc.history.JdbcSchemaHistory`.
+
```

Comment:
If this and similar properties have required values, can they be coded to carry the appropriate defaults?